services/src/pro/datasets/postgres.rs

```diff
@@ -5,114 +5,599 @@ use crate::datasets::storage::{
+            geoengine_operators::error::Error::LoadingInfo {
+                source: Box::new(e),
+            }
+        })?;
+        let stmt = conn
+            .prepare(
+                "
+        SELECT 
+            meta_data
+        FROM 
+            datasets 
+        WHERE 
+            id = $1",
+            )
+            .await
+            .map_err(|e| geoengine_operators::error::Error::LoadingInfo {
+                source: Box::new(e),
+            })?;
+
+        let row = conn.query_one(&stmt, &[&id]).await.map_err(|e| {
+            geoengine_operators::error::Error::LoadingInfo {
+                source: Box::new(e),
+            }
+        })?;
+
+        let meta_data: StaticMetaData<
+            OgrSourceDataset,
+            VectorResultDescriptor,
+            VectorQueryRectangle,
+        > = serde_json::from_value(row.get(0))?;
+
+        Ok(Box::new(meta_data))
+    }
 }
 
 #[async_trait]
-impl DatasetStore<UserSession> for PostgresDatasetDb {
+impl<Tls> MetaDataProvider<GdalLoadingInfo, RasterResultDescriptor, RasterQueryRectangle>
+    for PostgresDatasetDb<Tls>
+where
+    Tls: MakeTlsConnect<Socket> + Clone + Send + Sync + 'static,
+    <Tls as MakeTlsConnect<Socket>>::Stream: Send + Sync,
+    <Tls as MakeTlsConnect<Socket>>::TlsConnect: Send,
+    <<Tls as MakeTlsConnect<Socket>>::TlsConnect as TlsConnect<Socket>>::Future: Send,
+{
+    async fn meta_data(
+        &self,
+        dataset: &DatasetId,
+    ) -> Result<
+        Box<dyn MetaData<GdalLoadingInfo, RasterResultDescriptor, RasterQueryRectangle>>,
+        geoengine_operators::error::Error,
+    > {
+        let id = dataset
+            .internal()
+            .ok_or(geoengine_operators::error::Error::LoadingInfo {
+                source: Box::new(Error::InvalidDatasetId),
+            })?;
+
+        let conn = self.conn_pool.get().await.map_err(|e| {
+            geoengine_operators::error::Error::LoadingInfo {
+                source: Box::new(e),
+            }
+        })?;
+        let stmt = conn
+            .prepare(
+                "
+        SELECT 
+            meta_data
+        FROM 
+            datasets 
+        WHERE 
+            id = $1",
+            )
+            .await
+            .map_err(|e| geoengine_operators::error::Error::LoadingInfo {
+                source: Box::new(e),
+            })?;
+
+        let row = conn.query_one(&stmt, &[&id]).await.map_err(|e| {
+            geoengine_operators::error::Error::LoadingInfo {
+                source: Box::new(e),
+            }
+        })?;
+
+        // try if meta data is regular...
+        let meta_data: Result<GdalMetaDataRegular, _> = serde_json::from_value(row.get(0));
+
+        if let Ok(meta_data) = meta_data {
+            return Ok(Box::new(meta_data));
+        }
+
+        // ... or static.
+        let meta_data: GdalMetaDataStatic = serde_json::from_value(row.get(0))?;
+        dbg!(&meta_data);
+        Ok(Box::new(meta_data))
```

Comment:
Isn't this an enum that we could deserialize?

Moreover, please remove the `dbg!`

Reply:
9b7d107295722dc5fc39d86cedaae2c509438a0a